### PR TITLE
[JS] Fix ConcurrentModificationException scanning export class {} with multiple members

### PR DESCRIPTION
**Pull request by matthiasblaesing**
Fix applies same pattern, that is used in other places where ModelUtils#moveProperty is invoked.

### CODE REVIEW

**Comment by mbien** [comment]
> Fix applies same pattern, that is used in other places where ModelUtils#moveProperty is invoked.

i see one change in ModelVisitor. This sentence sounds like there should be multiple fixes in this PR?

**Comment by matthiasblaesing** [comment]
> > Fix applies same pattern, that is used in other places where ModelUtils#moveProperty is invoked.
> 
> i see one change in ModelVisitor. This sentence sounds like there should be multiple fixes in this PR?

Missunderstanding, I meant, that other location with similar structure, already hold the fix for `ConcurrentModificationException`:

https://github.com/apache/netbeans/blob/788ba1f72534348ac9e45210c7edd2428659d2b4/webcommon/javascript2.model/src/org/netbeans/modules/javascript2/model/JsObjectImpl.java#L488-L491

https://github.com/apache/netbeans/blob/788ba1f72534348ac9e45210c7edd2428659d2b4/webcommon/javascript2.model/src/org/netbeans/modules/javascript2/model/ModelVisitor.java#L1014-L1017

https://github.com/apache/netbeans/blob/788ba1f72534348ac9e45210c7edd2428659d2b4/webcommon/javascript2.model/src/org/netbeans/modules/javascript2/model/api/ModelUtils.java#L1917-L1920

**Review by mbien — APPROVED** [review]

File: webcommon/javascript2.model/src/org/netbeans/modules/javascript2/model/ModelVisitor.java [hunk 1]
```diff
@@ -712,8 +712,9 @@ public boolean enterClassNode(ClassNode node) {
             );
             JsObject origClassObject = parent.getProperty(className.getName());
             if (origClassObject != null) {
-                for (Entry<String, ? extends JsObject> e : origClassObject.getProperties().entrySet()) {
-                    ModelUtils.moveProperty(classObject, e.getValue());
+                List<JsObject> properties = new ArrayList<>(origClassObject.getProperties().values());
```
Comment (lbownik) [inline webcommon/javascript2.model/src/org/netbeans/modules/javascript2/model/ModelVisitor.java:715]: is there a specific reason for creating a new ArrayList ?

**Comment by neilcsmith-net** [comment]
Thanks @matthiasblaesing Adding label pending discussion on dev@  Will only go into delivery if we have an 18-rc5.  Otherwise, should go in to master and be a possible cherry pick candidate to release180 for an 18-u1.

**Comment by neilcsmith-net** [comment]
OK, moving this on to master so that we can get rid of the delivery branch.

**Pull request merged**